engine/storage/kv/event.go

```diff
@@ -62,13 +66,21 @@ func (es *kvEventSubscription) get(ctx context.Context, b kv.Bucket, name string
 	es.Event = workflow.EventFlag(eventFlag).String()
 	if ok, err := b.Has(ctx, name+keySfxEventContext); err != nil {
 		return fmt.Errorf("checking event context: %w", err)
-	} else if !ok {
-		return nil
+	} else if ok {
+		if ctxBytes, err := b.Get(ctx, name+keySfxEventContext); err != nil {
+			return fmt.Errorf("getting event context: %w", err)
+		} else {
+			es.Context = string(ctxBytes)
+		}
 	}
-	if ctxBytes, err := b.Get(ctx, name+keySfxEventContext); err != nil {
-		return fmt.Errorf("getting event context: %w", err)
-	} else {
-		es.Context = string(ctxBytes)
+	if ok, err := b.Has(ctx, name+keySfxEventEventContext); err != nil {
+		return fmt.Errorf("checking event event_context: %w", err)
+	} else if ok {
+		if evCtxBytes, err := b.Get(ctx, name+keySfxEventEventContext); err != nil {
+			return fmt.Errorf("getting event event_context: %w", err)
+		} else {
+			es.EventContext = string(evCtxBytes)
+		}
```

Comment:
Will we be extracting bytes like this ever again? Should we turn this into a function that we pass bytes into along with a key name?

If that doesn't make sense for this, then all good :D Just saw the pattern.

Reply:
Uhm, maybe? For this particular bucket it's just the context and the event context, I think. Yeah could go either way. If we added a third field that worked like this that'd probably be the time. :)

Reply:
Also fwiw I'd like to generalize the kv bucket abstraction to include a standardized error for a key-not-found error. This would make the operation more atomic and wouldn't need the (race condition prone) Has/Get pattern.